File: smart-hub-direct-kt/src/main/kotlin/io/github/tassiLuca/hub/Launcher.kt
package io.github.tassiLuca.hub

import io.github.tassiLuca.hub.adapters.MockedHubManager
import kotlinx.coroutines.runBlocking

/** The application entry point. */
fun main(): Unit = runBlocking {
    MockedHubManager(this.coroutineContext).run()
}

File: smart-hub-direct/src/main/scala/io/github/tassiLuca/hub/Launcher.scala
package io.github.tassiLuca.hub

import gears.async.default.given
import gears.async.Async
import io.github.tassiLuca.hub.adapters.MockedHubManager

/** The launcher of a mocked version of the application, using UI simulators. */
@main def launchUIMockedHub(): Unit = Async.blocking:
  MockedHubManager().run()
